src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -97,9 +99,10 @@ protected bool InitializeFileWatcherIfEnabled()
         {
             if (Host.ScriptConfig.FileWatchingEnabled)
             {
-                string functionScriptDirectory = Path.GetDirectoryName(Metadata.ScriptFile);
-                _fileWatcher = new AutoRecoveringFileSystemWatcher(functionScriptDirectory);
-                _fileWatcher.Changed += OnScriptFileChanged;
+                _fileChangeSubscription = Host.EventManager.OfType<FileEvent>()
+                    .Where(f => string.Equals(f.Source, "ScriptFiles", StringComparison.Ordinal) &&
+                    f.FileChangeArguments.Name.StartsWith(Metadata.Name, StringComparison.OrdinalIgnoreCase))
```

Comment:
_I somehow lost this comment in my review_

The previous FileWatcher matched on the script filename. This new subscription is looking at `StartsWith(functionName)`. Could that have some false matches if I had two functions named `MyFunction` and `MyFunction1`?